R/GatingSet2flowJo.R

```diff
@@ -270,17 +304,39 @@ transformationNode <- function(gh, matInfo){
                   )
 
                 }else if(trans.type == "logicle"){
-                  param <- as.list(environment(func))
-                  withBasis <- - 10 ^ (2 * as.vector(param[["w"]]))
-                  transNode <- xmlNode("biex"
-                                       , namespace = "transforms"
-                                       , attrs = c("transforms:length" = 4096
-                                                   , "transforms:maxRange" = param[["t"]]
-                                                   , "transforms:neg" = param[["a"]]
-                                                   , "transforms:width" = withBasis
-                                                   , "transforms:pos" = param[["m"]]
-                                       )
-                  )
+
+                  if(!is.null(Scaling)){
+                  # Added scaling adjustment
+                    offset <- 316
+                    decades <- 3162277.7
+
+                    param.loc <- which(gsub("Comp-", "", chnl) == names(Scaling))
+                    # param.loc <- grep(chnl, names(Scaling))
+                    if (length(param.loc) == 1){
+                        offset <- Scaling[[param.loc]][1]
+                        decades <- log10(Scaling[[param.loc]][2])-log10(offset)
+                    }
+                                      # decades <- log10(3162277.7/2)-2
+                    transNode <- xmlNode("log"
```

Comment:
Is it appropriate to convert logicle to log transformation?

Reply:
This is an interesting point. I can't remember if I thought about this and ignored it or if it never occured to me. We usually just use biex transforms in our analysis. But the particular project I was working on needed a different transform. It was also a bit of work to choose a transformation that I could get into flowWorkspace/CytoML/FlowJo. In this case having a "logicle" and changing it to log for FlowJo did not cause any changes (at least any major or medium changes). Again, I didn't mean for this commit to be computationally flawless.